### PR TITLE
docs: add explorer api docs and query builder ui docs

### PR DESCRIPTION
**Pull request by ctrombley**
**DO NOT MERGE** until the query builder GA release.

### What
This PR adds API docs for the Explorer GA.

- [JIRA](https://hashicorp.atlassian.net/browse/TF-11368)

### Why
These updates support the release of the query builder feature for Explorer and set up the Explorer GA release.

#### Explorer API
Screenshot 2023-12-18 at 4 37 58 PM

#### Workspace Health
Screenshot 2024-01-24 at 1 08 35 PM

### Merge Checklist
_If items do not apply to your changes, add (N/A) and mark them as complete._

#### Pull Request
- [x] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. tfc).
- [x] Description links to related pull requests or issues, if any.

#### Content
- [x] API documentation and the API Changelog have been updated. 
- [x] Links to related content where appropriate (e.g., API endpoints, permissions, etc.).
- [x] Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages.
- [x] New pages have metadata (page name and description) at the top.
- [x] New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars.
- [x] The Vercel website preview successfully deployed.
- [x] (N/A) Redirects have been added to `website/redirects.js` for moved, renamed, or deleted pages.
- [x] (N/A) Pages with related content are updated and link to this content when appropriate.
- [x] (N/A) New images are 2048 px wide. They have HashiCorp standard annotation color (#F92672) and format (rectangle with rounded corners), blurred sensitive details (e.g. credentials, usernames, user icons), and descriptive alt text in the markdown for accessibility.
- [x] (N/A) UI elements (button names, page names, etc.) are bolded.

#### Reviews
- [x] I or someone else reviewed the content for technical accuracy.
- [x] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

### CODE REVIEW

File: website/docs/cloud-docs/api-docs/explorer.mdx [hunk 1]
```diff
@@ -0,0 +1,460 @@
...
+| `is_empty`         | `boolean`, `number`, `string` | Filters data to rows where the field does not contain a value. |
+| `is_not_empty`     | `boolean`, `number`, `string` | Filters data to rows where the field contains any value. |
+| `gt`               | `number`                      | Filters data to rows where the field value is greater than the filter value. |
+| `lt`               | `number`                      | Filters data to rows where the field value is less than the filter value. |
+| `gteq`             | `number`                      | Filters data to rows where the field value is greater than or equal to the filter value. |
+| `lteq`             | `number`                      | Filters data to rows where the field value is less than or equal to the filter value. |
+| `is_before`        | `datetime`                    | Filters data to rows where the field value is earlier than the filter value. ISO 8601 formatted dates are recommended for filter values. If no time zone offset is provided, the filter value will be interpreted as UTC. |
```
Comment (BrianMMcClain) [inline website/docs/cloud-docs/api-docs/explorer.mdx:241]: Are other options than ISO 8601 formatted dates supported? As best I can tell, we're pretty prescriptive that dates and times are all in ISO 8601 format, but just want to make sure we're being exact here.

File: website/docs/cloud-docs/workspaces/explorer.mdx [hunk 2]
```diff
@@ -56,3 +56,38 @@ The explorer for workspace visibility provides the following queries for specifi
...
+
+Finally, the value field allows you to provide a value for the condition.
+
+-> **Tip:** Inspect the filter conditions used by the various pre-canned [**Use
+Cases**](#use-cases) to learn how they are constructed.
+
+Multiple filter conditions can be used within a query. When multiple conditions
+are provided, they are evaluated at query time with a logical AND. To add a new
+condition, use the "Add condition" button below the condition list. To remove a
+condition, use the trash bin button on the right hand side of the condition.
```
Comment (BrianMMcClain) [inline website/docs/cloud-docs/workspaces/explorer.mdx:88]:
```suggestion
Multiple filter conditions can be used within a query. When multiple conditions
are provided, they are evaluated at query time with a logical AND. To add a new
condition, use the **Add condition** button below the condition list. To remove a
condition, use the trash bin button on the right hand side of the condition.
```

File: website/docs/cloud-docs/workspaces/explorer.mdx [hunk 3]
```diff
@@ -56,3 +56,38 @@ The explorer for workspace visibility provides the following queries for specifi
...
+
+While you stage updates to filter conditions using the query builder, query
+results will not be re-evaluated automatically. To execute the query once
+the desired filter conditions have been defined, press the "Run Query" button at
+the bottom of the query builder.
```
Comment (BrianMMcClain) [inline website/docs/cloud-docs/workspaces/explorer.mdx:93]:
```suggestion
While you stage updates to filter conditions using the query builder, query
results will not be re-evaluated automatically. To execute the query once
the desired filter conditions have been defined, press the **Run Query** button at
the bottom of the query builder.
```

File: website/docs/cloud-docs/workspaces/explorer.mdx [hunk 4]
```diff
@@ -56,3 +56,38 @@ The explorer for workspace visibility provides the following queries for specifi
...
 
+## Custom filter conditions
+
+Explorer's query builder allows you to execute queries with custom filter
+conditions against any of the supported [**Types**](#types).
+
+To get started, select a [**Type**](#types) or [**Use Case**](#use-cases) from the
+Explorer home page. You can expand the "Modify conditions" section to show the
+filter conditions in use for the current query and to define new filter
+conditions.
```
Comment (BrianMMcClain) [inline website/docs/cloud-docs/workspaces/explorer.mdx:67]:
```suggestion
To get started, select a [**Type**](#types) or [**Use Case**](#use-cases) from the
Explorer home page. You can expand the **Modify conditions** section to show the
filter conditions in use for the current query and to define new filter
conditions.
```

**Review by BrianMMcClain — CHANGES_REQUESTED** [review]

File: website/docs/cloud-docs/api-docs/explorer.mdx [hunk 5]
```diff
@@ -0,0 +1,460 @@
...
+| `modules`                         | `string`   | A comma separated list of modules used by this workspace. |
+| `organization_name`               | `string`   | The name of the organization this workspace belongs to. |
+| `project_external_id`             | `string`   | The external ID of the project this workspace belongs to. |
+| `project_name`                    | `string`   | The name of the project this workspace belongs to.
+| `provider_count`                  | `number`   | The number of distinct providers used in this workspace. |
+| `providers`                       | `string`   | A comma separated list of providers used in this workspace. |
+| `resources_drifted`               | `number`   | The count of resources that drift was detected for. |
+| `resources_undrifted`             | `number`   | The count of resources that drift was not detected for. |
+| `state_version_terraform_version` | `string`   | The Terraform version used to create the current run's state file. |
+| `vcs_repo_identifier`             | `string`   | The Terraform version used by this workspace. |
```
[the remark this answers is not in the document]
Comment (ctrombley) [reply website/docs/cloud-docs/api-docs/explorer.mdx:125]: Good catch!

File: website/docs/cloud-docs/api-docs/explorer.mdx [hunk 6]
```diff
@@ -0,0 +1,460 @@
...
+| `workspace_created_at`            | `datetime` | The time this workspace was created. |
+| `workspace_name`                  | `string`   | The name of this workspace. |
+| `workspace_terraform_version`     | `string`   | The Terraform version currently configured for this workspace. |
+| `workspace_updated_at`            | `datetime` | The time this workspace was last updated. |
+| `workspace_resource_count`        | `number`   | The time this workspace was last updated. |
```
[the remark this answers is not in the document]
Comment (ctrombley) [reply website/docs/cloud-docs/api-docs/explorer.mdx:130]: Good catch, thanks!

File: website/docs/cloud-docs/api-docs/explorer.mdx (hunk 1 above)
Comment (ctrombley) [reply website/docs/cloud-docs/api-docs/explorer.mdx:240]: I think that rails is able to parse timestamps with slight differences, but I see no reason not to use more exact language.

**Review by BrianMMcClain — APPROVED** [review]
LGTM! Just pushed one small change to fix the table formatting and a merge conflict

File: website/docs/cloud-docs/api-docs/explorer.mdx [hunk 7]
```diff
@@ -0,0 +1,459 @@
...
+There are several contextual views available for querying:
+
+| Type          | Description |
+| ------------- | ----------------------------------------------------------------------------- |
+| `workspaces`  | Information about the workspaces in the target organization and any current runs associated with them. Each row returned represents a single workspace. See [View Type: workspaces](#view-type-workspaces) for available fields. |
+| `tf_versions` | Information about the Terraform versions in use across this organization. Each row returned represents a Terraform version in use. See [View Type: tf_versions](#view-type-tf_versions) for available fields. |
+| `providers`   | Information about the Terraform providers in use across the target organization. Each row returned represents a Terraform provider in use. See [View Type: providers](#view-type-providers) for available fields. |
+| `modules`     | Information about the Terrraform modules in use across the target organization. Each row returned represents a Terraform module in use. See [View Type: modules](#view-type-modules) for available fields. |
```
Comment (simonxmh) [inline website/docs/cloud-docs/api-docs/explorer.mdx:96]:
```suggestion
| `modules`     | Information about the Terraform modules in use across the target organization. Each row returned represents a Terraform module in use. See [View Type: modules](#view-type-modules) for available fields. |
```

File: website/docs/cloud-docs/api-docs/explorer.mdx [hunk 8]
```diff
@@ -0,0 +1,459 @@
...
+| `providers`                       | `string`   | A comma separated list of providers used in this workspace. |
+| `resources_drifted`               | `number`   | The count of resources that drift was detected for. |
+| `resources_undrifted`             | `number`   | The count of resources that drift was not detected for. |
+| `state_version_terraform_version` | `string`   | The Terraform version used to create the current run's state file. |
+| `vcs_repo_identifier`             | `string`   | The name of the repository configured for this workspace, if available. |
+| `workspace_created_at`            | `datetime` | The time this workspace was created. |
+| `workspace_name`                  | `string`   | The name of this workspace. |
+| `workspace_terraform_version`     | `string`   | The Terraform version currently configured for this workspace. |
+| `workspace_updated_at`            | `datetime` | The time this workspace was last updated. |
+| `workspace_resource_count`        | `number`   | The count of resources in
```
Comment (simonxmh) [inline website/docs/cloud-docs/api-docs/explorer.mdx:130]: this field does not get returned right now

File: website/docs/cloud-docs/api-docs/explorer.mdx [hunk 9]
```diff
@@ -0,0 +1,459 @@
...
+| `workspace_count` | `number` | The number of workspaces using this provider. |
+| `workspaces`      | `string` | A comma-separated list of workspaces using this provider. |
+
+#### View Type: `modules`
+
+| Field             | Type     | Description |
+| ------------------| -------- | ------------------------------------------------------- |
+| `name`            | `string` | The name of this module. |
+| `source`          | `string` | The source of this module. |
+| `version`         | `string` | The semantic version string for this module. |
```
Comment (simonxmh) [inline website/docs/cloud-docs/api-docs/explorer.mdx:157]: We're missing the `registry_type` field here

File: website/docs/cloud-docs/api-docs/explorer.mdx [hunk 10]
```diff
@@ -0,0 +1,459 @@
...
+### View Types
+
+Explorer queries each require a view, specified by the `type` query string parameter.
+There are several contextual views available for querying:
+
+| Type          | Description |
+| ------------- | ----------------------------------------------------------------------------- |
+| `workspaces`  | Information about the workspaces in the target organization and any current runs associated with them. Each row returned represents a single workspace. See [View Type: workspaces](#view-type-workspaces) for available fields. |
+| `tf_versions` | Information about the Terraform versions in use across this organization. Each row returned represents a Terraform version in use. See [View Type: tf_versions](#view-type-tf_versions) for available fields. |
+| `providers`   | Information about the Terraform providers in use across the target organization. Each row returned represents a Terraform provider in use. See [View Type: providers](#view-type-providers) for available fields. |
```
Comment (simonxmh) [inline website/docs/cloud-docs/api-docs/explorer.mdx:95]: we might need to clarify what "in use" means, perhaps the fact they need have had a run with the new providers for it to be reflected?

File: website/docs/cloud-docs/api-docs/explorer.mdx [hunk 11]
```diff
@@ -0,0 +1,459 @@
...
+
+## Export data as CSV
+
+`GET /organizations/:organization_name/explorer/export/csv`
+
+| Parameter            | Description |
+| -------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `:organization_name` | The name of the organization to query. The organization must already exist in the system, and the user must have permissions to read the workspaces and projects within it. |
+
+This endpoint can be used to download a full, unpaged export of the query results
```
Comment (simonxmh) [inline website/docs/cloud-docs/api-docs/explorer.mdx:420]: Can we also specify that the filters and sorts will also be applied to the export result? 
but the pagination will not (that's covered here)

File: website/docs/cloud-docs/api-docs/explorer.mdx [hunk 12]
```diff
@@ -0,0 +1,459 @@
...
+in CSV format.
+
+Explorer queries will time out after 60 seconds. If the desired query is timing
+out, try a simpler query that uses less filters.
+
+Requests to the explorer are subject to [rate
```
Comment (simonxmh) [inline website/docs/cloud-docs/api-docs/explorer.mdx:426]: The rate limits should be a stand alone section?

**Review by simonxmh — COMMENTED** [review]
Really nice additions! Thank you so much! A few suggestions here and there

File: website/docs/cloud-docs/api-docs/explorer.mdx [hunk 13]
```diff
@@ -0,0 +1,459 @@
...
+## Export data as CSV
+
+`GET /organizations/:organization_name/explorer/export/csv`
+
+| Parameter            | Description |
+| -------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `:organization_name` | The name of the organization to query. The organization must already exist in the system, and the user must have permissions to read the workspaces and projects within it. |
+
+This endpoint can be used to download a full, unpaged export of the query results
+in CSV format. (with the filter, sort, and field selections applied)
```
Comment (BrianMMcClain) [inline website/docs/cloud-docs/api-docs/explorer.mdx:421]:
```suggestion
in CSV format (with the filter, sort, and field selections applied).
```

**Review by BrianMMcClain — APPROVED** [review]
Just one small fix, otherwise LGTM!

**Pull request merged**